### DISCUSSION TITLE
Seplos BMS 3.0 Help

### DISCUSSION THREAD

**Discussion by csw03** [syssi/esphome-seplos-bms#120] [Q&A] [answered]
Hello
i try different solution yaml file, different 485 boards but nothing can't comunicate with BMS

SEPLOS 3.0 Type C SPB05

BMS Seplos 3.0 im using ESP32 DEVKIT V2 from AZ Delivery
TX GPIO1
RX GPIO3

My conf YAML

```
substitutions:
  name: "seplos-bms-3"
  device_description: "Monitor and control a Seplos V3 BMS via RS485 (Modbus)"
  tx_pin: GPIO1
  rx_pin: GPIO3

esphome:
  name: ${name}
  comment: ${device_description}
  min_version: 2024.6.0
  project:
    name: "syssi.esphome-seplos-bms"
    version: 1.2.0
 
esp32:
  board: esp32dev
  framework:
    type: esp-idf

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
  
  manual_ip:
    static_ip: 192.168.1.135
    gateway: 192.168.0.1
    subnet: 255.255.255.0




 
  ap:
    ssid: "Secondo-Esp Fallback Hotspot"
    password: "SpiXXXXXXXYQrQ"

captive_portal:

improv_serial:

web_server:
  port: 80  

   


ota:
  - platform: esphome
    password: "b785cXXXXXXXXXXXXXXXf6fc5de"



logger:
  level: DEBUG



api:
  encryption:
    key: "spQiuLNtXXXXXXXXXXXXXXXXXXXXXikwyA="



uart:
  - id: uart_0
    baud_rate: 19200
    tx_pin: ${tx_pin}
    rx_pin: ${rx_pin}
    debug:
      direction: BOTH
      dummy_receiver: false
    
    

modbus:
  - id: modbus0
    uart_id: uart_0
    send_wait_time: 200ms

modbus_controller:
  - id: bms0
    address: 0x00
    modbus_id: modbus0
    command_throttle: 200ms
    update_interval: 10s

sensor: /.............................................../
```

[Communications Records 20241221_1305.txt](https://github.com/user-attachments/files/18218475/Communications.Records.20241221_1305.txt)
![webIP](https://github.com/user-attachments/assets/2e06ca5d-91d6-40a0-9c68-aeec8cbb8bed)

**Comment by syssi** [comment]
If you use `GPIO1/GPIO3` you have to set the `baud_rate` of the `logger` component to `0`. Could you provide a photo of your wiring?

**Comment by csw03** [comment]
i have correct yaml to baud_rate 0 of the logger, but nothing, here the rs485 module i using now
![22f9a216-207d-436f-be4e-789edcdccc23](https://github.com/user-attachments/assets/880c3a12-825b-48d6-bd58-7c92b51418db)
![46508372-e186-4a15-be83-cddbef607ec9](https://github.com/user-attachments/assets/63b3f993-835b-4f96-83d5-8a956fbf6f9c)
![f6962cf4-9a74-4d6f-9b5d-c923bd72b250](https://github.com/user-attachments/assets/df52d3f0-2372-435a-8664-82fffb0f6486)

**Reply by syssi** [reply, marked as answer]
Your wiring isn't fine. Your RS485 requires flow control. It should look like this: https://github.com/syssi/esphome-solax-x1-mini?tab=readme-ov-file#rs485-ttl-module-with-flow-control-pin

The `modbus` section must be extended to control the flow:

```
modbus:
  - id: modbus0
    uart_id: uart_0
    send_wait_time: 200ms
    flow_control_pin: GPIO0
```

**Comment by csw03** [comment]
Finally all working good with GPIO0 flow control
Thank you so much for your help
![seplos v3 work](https://github.com/user-attachments/assets/c16e1ebf-d253-4ff7-911c-eaa9ccce7e3b)

**Reply by syssi** [reply]
Good job!

**Comment by csw03** [comment]
Maybe you know the address for can add the cell delta voltage ?
Thank you for your work

**Reply by syssi** [reply]
There is no dedicated register for the delta cell voltage but can calculate the delta using a template sensor by substract the `min cell voltage` from the `max cell voltage`:

https://github.com/syssi/esphome-seplos-bms/blob/main/docs/XZH%20BMS%20Modbus-RTU%20Protocol.pdf
https://github.com/syssi/esphome-seplos-bms/blob/main/esp8266-seplos-v3-example.yaml#L192-L218

**Reply by csw03** [reply]
Perfect, thank you
